spark-jobs/src/main/scala/za/co/absa/enceladus/common/CommonJobExecution.scala

```diff
@@ -52,20 +54,26 @@ trait CommonJobExecution extends ProjectMetadata {
                                          performance: PerformanceMeasurer)
 
   TimeZoneNormalizer.normalizeJVMTimeZone()
-  SparkVersionGuard.fromDefaultSparkCompatibilitySettings.ensureSparkVersionCompatibility(SPARK_VERSION)
+  SparkVersionGuard.fromSpark3XCompatibilitySettings.ensureSparkVersionCompatibility(SPARK_VERSION)
```

Comment:
Don't we want to ensure the Spark is at least 3.2.1?

Reply:
We can do that, but I changed to using the `SparkVersionGuard` from spark-commons, as I noticed we still had one in Enceladus. `fromSpark3XCompatibilitySettings` has 3.0.0-4.0.0 boundaries . We can instantiate a `SparkVersionGuard` manually if it makes sense